hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java

```diff
@@ -81,7 +81,12 @@ class AbstractDelegationTokenSecretManager<TokenIdent
       = DelegationTokenSecretManagerMetrics.create();
 
   private String formatTokenId(TokenIdent id) {
-    return "(" + id + ")";
+    try {
+      return "(" + id + ")";
+    } catch (Exception e) {
+      LOG.warn("Exception in formatTokenId", e);
```

Comment:
What can this exception be?

Reply:
I'm sorry I reply late, here will generate java. lang. IllegalArgumentException

Reply:
As HDFS-17138 mentioned, here will throw NoMatchingRule if set unexpected kerberos rules.
https://github.com/apache/hadoop/blob/8d95c588d2df0048b0d3eb711d74bf34bf4ae3c4/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/util/KerberosName.java#L428-L430